システム開発ガイドより、プロジェクト固有とする必要がない差分内容を反映

tisuchida opened the pull request:
https://github.com/nablarch-development-standards/nablarch-development-standards/pull/41
https://github.com/nablarch-development-standards/nablarch-development-standards/pull/42
と合わせ、システム開発ガイドと開発標準を比較した際に出た差分のうち、開発標準に取り込むべき内容を反映しました。

1.  UI標準(画面)

    -   全体
        -   ヘッダに、プロジェクト名・システム名を取得する数式を追加
    -   1.3.	見出し中の「JakartaEE」を「Jakarta EE」に変更
    -   3.2.1.2.	補足説明
        -   「検索条件入力画面」の説明に、以下を追加（システム開発ガイドの内容反映）
            -   検索結果が0件の場合、検索条件入力画面内でユーザにエラー通知する。
        -   「検索結果一覧画面」の説明に、以下を追加（システム開発ガイドの内容反映）
            -   詳細画面から戻る際に、前回の検索条件を復元して再検索を行う。
        -   「詳細画面」の説明に、以下を追加（システム開発ガイドの内容反映）
            -   詳細画面から戻る際に、前回の検索条件を復元して再検索を行う。
    -   3.2.4.	「検索(検索一覧) → 単一更新・削除」型
        -   システム開発ガイドの記載で置換
    -   3.2.5.2	補足説明
        -   「検索結果一覧更新/削除画面」の説明に、以下を追加（システム開発ガイドの内容反映）
            -   検索結果はページングできるようにする。
        -   「一括更新/一括削除完了画面」を「一括更新/削除完了画面」に修正（システム開発ガイドの内容反映）
    -   3.4.	エラー時の画面遷移
        -   表の欄外に、以下の説明を記載（システム開発ガイドの内容反映）
            -   ※排他制御エラーについて、本システムではデータの同時更新が頻繁に発生することは想定されないため、元画面への遷移は行わなず共通エラー画面への遷移で対処することとする。
    -   5.5.3.	戻る機能への対応方針
        -   ブラウザバックを、「許容する」から「動作保証対象外とする」に記載変更（システム開発ガイドの内容反映）
    -   5.7.	メッセージエリア
        -   表中「業務画面に遷移するパターン」の用途説明を変更（システム開発ガイドの内容反映）
            -   単項目精査エラーの通知
                **フォームで完結しない精査エラーの通知（単項目に対するエラーの場合）**
            -   項目間精査エラーの通知
                **フォームで完結しない精査エラーの通知（複数項目に対するエラーの場合）**

2.  UI標準(画面)別冊\_UI部品カタログ

    -   全体
        -   ヘッダに、プロジェクト名・システム名を取得する数式を追加
    -   1.3.	見出し中の「JakartaEE」を「Jakarta EE」に変更
    -   3.1.2.	複数行テキスト入力
        -   (3)機能 (3-1)通常時 (a)入力時 の記載を修正
            -   入力欄の縦と横の幅を <del>変更</del> **設定** 可能である。表示幅以上の文字を入力した場合、入力項目内で縦スクロールが発生する。
    -   3.2.1.	チェックボックス
        -   (5)備考 (a)実装方法の採用指針 の記載を修正（システム開発ガイドの内容反映）
            -   どれを <del>選択</del> **採用** すべきかは、下記３つの観点をもとに判断する。
            -   コード値の使用 <del>可否</del> **有無**
    -   3.2.2.	ラジオボタン
        -   (5)備考 (a)実装方法の採用指針 の記載を修正（システム開発ガイドの内容反映）
            -   Nablarchではラジオボタンの用途に応じて、 <del>4</del> **3** 種類の実装方法を推奨している。
            -   採用指針は全部で <del>3</del> **2** 種類ある。
            -   コード値の使用 <del>可否</del> **有無**
            -   下記に用途に応じた <del>6</del> **3** 種類の実装方法を示す。
    -   3.2.3.	プルダウンメニュー
        -   (3)機能 (3-2)編集不可 の記載を修正（システム開発ガイドの内容反映）
            -   <del>選択有無に関わらず、</del> **サブミット時は** サーバ側に値 <del>は</del> **を** 送信しない。
    -   3.2.4.	リストボックス
        -   (3)機能 の項目名を以下のように修正（システム開発ガイドの内容反映）
            -   (3-2) 編集不可(disabled) <del>の場合</del>
            -   (3-3) 読み取り専用(readonly) <del>の場合</del>
        -   (3)機能 (3-2)編集不可(disabled) の記載を修正（システム開発ガイドの内容反映）
            -   選択有無に関わらず、 <del>候補</del> **選択肢** をマウス等で押下しても選択状態は切り替わらない。

3.  各単体テスト仕様書フォーマット共通

    -   「【テストターゲット名】\_クラス単体」シートのケースNoカラムヘッダに、採番ルールコメントを追加
    -   「【リクエストID】\_リクエスト単体」シートのケースNoカラムヘッダに、採番ルールコメントを追加
    -   「取引単体」シートのケースNoカラムヘッダに、採番ルールコメントを追加
    -   「取引単体」シートのケースNoカラムに、数式を設定

4.  単体テスト仕様書(バッチ)\_(取引ID)\_(取引名)

    -   「【バッチ処理ID】\_リクエスト単体」シートの名前を「【リクエストID】\_リクエスト単体」に変更
    -   「【リクエストID】\_リクエスト単体」シートヘッダの「バッチ処理ID」を「リクエストID」に変更

5.  単体テスト仕様書(共通コンポーネント)\_(共通コンポーネントID)\_(共通コンポーネント名)

    -   「【テスト対象コンポーネント名】」シートの名前を「【テストターゲット名】\_クラス単体」に変更

6.  各単体テスト仕様書サンプル共通

    -   フォーマットの変更を反映
    -   クラス単体シートの「【テストターゲット名】」を、本文ヘッダに記載しているテストターゲット名に置換
    -   リクエスト単体シートの「【リクエストID】」を、本文ヘッダに記載しているリクエストIDに置換

7.  DB\_Design\_Standards

    -   3.2.7.1 Logical name
        -   記載を以下の通り変更（システム開発ガイドの内容反映）
            -   No definition required (because there is no need to **manage** unique keys with logical names)
    -   3.2.7.3 Naming example
        -   オブジェクト内の以下の記載を変更（システム開発ガイドの内容反映、タイポの修正）
            -   uk\_system\_ac **c** ount\_01

tisuchida committed:
システム開発ガイドより、プロジェクト固有とする必要がない差分内容を反映

tisuchida committed:
各単体テスト仕様書のシート名・ヘッダを調整

tisuchida commented:
別途指摘をふまえ、以下の通り修正しました。

- 各単体テスト仕様書フォーマット
  - 「【リクエストID】クラス単体テスト」シート名を「【テストターゲット名】クラス単体に変更
  - 「【リクエストID】リクエスト単体テスト」シート名を「【リクエストID】リクエスト単体」に変更
  - 各リクエスト単体のシートヘッダの「テストターゲット名」の記載を削除し、代わりに「リクエストID」「バッチ処理ID」を追加

- 各単体テスト仕様書サンプル
  - フォーマットでの修正事項を反映
  - 各クラス単体のシート名に、テストターゲットとなるクラス名を記載
  - 各リクエスト単体のシート名に、リクエストIDを記載


664bece4cde3259f4ae05f538846349d03b9ccd9

tisuchida committed:
取引単体のケース番号を数式化

tisuchida committed:
リクエスト単体の「リクエストID」「バッチ処理ID」を「テストターゲット名」に変更

tisuchida committed:
単体テスト仕様書の観点誤りを修正

リクエスト単体の5-4-1を以下の通り修正。
（誤）「数値項目に有効範囲の下限値＋1が入力された場合、エラーとなること。」
（正）「数値項目に有効範囲の下限値－1が入力された場合、エラーとなること。」

tisuchida committed:
取引単体のケース番号数式を修正

kumagoro1202 commented on 030_設計ドキュメント/010_フォーマット/110_テスト仕様書/単体テスト仕様書(画面)_(取引ID)_(取引名).xlsx:
リクエスト単体テストのヘッダーについて
口頭で会話した際にシステム開発ガイドと合わせればいいという話にはなりましたが、どう合わせるかは具体的に話せていなかったかと思います。
現在はリクエストIDを削除してシステム開発ガイドに合わせようとしてくれていますが、開発標準が対象とするシステムにおいてはリクエスト単体テストはリクエストID単位にシートを作成するものだと思います。（サンプルのリクエスト単体テストシートの上部オブジェクトにも書かれています。）リクエストIDは残し、システム開発ガイド側のフォーマットにリクエストIDを追加した上で、サンプルプロジェクトにおいては画面単位でリクエスト単体テストを行うため不要とかのコメントをしたほうが良いと思います。
↓この2つの修正は元に戻して欲しいです。
- 「【リクエストID】リクエスト単体」シート名を「【テストターゲット名】リクエスト単体」に変更
- 「【テストターゲット名】リクエスト単体」シートのヘッダにある「リクエストID」「バッチ処理ID」を「テストターゲット名」に変更

tisuchida replied:
質問が2点あります。
1. 別途指摘で、サンプルプロジェクトの画面の単体テスト仕様書については、シート名に画面のIDが記載されることを踏まえてフォーマット側のシート名を「【画面ID】_リクエスト単体」とすればよいとなったと思います。これに合わせて、ヘッダ部分も「リクエストID」を「画面ID」とするのが良いように思いますがいかがでしょうか。
2. サンプルプロジェクトの単体テスト仕様書は、Webサービスのものもあります。Webサービスのものについては、「リクエストID」に相当するものが「API ID」と設計書に記載されていますが、これも「API ID」としてしまってよいでしょうか。

kumagoro1202 replied:
1.  システム開発ガイドのサンプルプロジェクトはNTFを使わない前提なのでシートの単位が画面単位になるので画面IDにするという話になりました。対して開発標準はNTFを前提としているのでシートはリクエスト単位にします。そのためリクエストIDは変更しないでください。
2. 単体テスト仕様書はNTFとのひも付きが強いのでリクエストIDにしましょう。バッチ処理IDもリクエストIDにして単体テスト仕様書上では実行基盤に関わらずリクエストIDの呼称を使いましょう。

tisuchida replied:
以下の修正を実施しました。

- フォーマット
  - リクエスト単体のシート名を、「【テストターゲット名】リクエスト単体」から「【リクエストID】リクエスト単体」に戻しました。
  - リクエスト単体シートのヘッダの「テストターゲット名」の1行下に「リクエストID」を追加しました。
  - バッチのリクエスト単体については、「バッチ処理ID」の記載を「リクエストID」に修正しました。
- サンプル
  - フォーマットの修正を取り込みました。

tisuchida committed:
リクエスト単体のヘッダに「リクエストID」「バッチ処理ID」を追加

tisuchida committed:
フォーマットの微修正

- 単体テスト仕様書(バッチ)の、「バッチ処理ID」を「リクエストID」に変更
- リクエスト単体シートのシート名を「テストターゲット名」から「リクエストID」に変更
